platform/posix/transport/src/openssl_posix.c

```diff
@@ -590,10 +617,8 @@ OpensslStatus_t Openssl_Disconnect( NetworkContext_t * pNetworkContext )
     /* Tear down the socket connection. */
     socketStatus = Sockets_Disconnect( pNetworkContext->socketDescriptor );
 
-    if( socketStatus == SOCKETS_INVALID_PARAMETER )
-    {
-        returnStatus = OPENSSL_INVALID_PARAMETER;
-    }
+    /* Convert socket wrapper status to openssl status. */
```

Comment:
I see a NULL de-referencing bug in this function where a NULL `pNetworkContext` will be de-referenced in the call to `Sockets_Disconnect`

Reply:
Yea. That's a good find. Unrelated to this change though, but I think it definitely will be a good fix. I will fix it. Thanks.

Reply:
Suggest refactoring for readability and to remove the possible `NULL` dereference at line 618:

```c
    SocketStatus_t socketStatus = SOCKETS_INVALID_PARAMETER:

    if( pNetworkContext != NULL )
    {
        if( pNetworkContext->pSsl != NULL )
        {
            /* SSL shutdown logic in here... */
        }
         /* Tear down the socket connection, pNetworkContext != NULL here. */ 
        socketStatus = Sockets_Disconnect( pNetworkContext->socketDescriptor );
    }
    else
    {
        LogError( ( "Parameter check failed: pNetworkContext is NULL." ) );
    }
    return convertToOpensslStatus( socketStatus );
```

Reply:
Thanks Gary. Addressed it.